crates/ruff_linter/src/rules/flake8_use_pathlib/rules/os_getcwd.rs

```diff
@@ -85,8 +89,12 @@ pub(crate) fn os_getcwd(checker: &Checker, call: &ExprCall, segments: &[&str]) {
 
             let applicability = if checker.comment_ranges().intersects(range) {
                 Applicability::Unsafe
-            } else {
+            } else if is_top_level_expression_call(checker, call) {
+                // Safe when the call is a top-level expression (return value not used)
                 Applicability::Safe
+            } else {
+                // Unsafe because the return type changes (str/bytes -> Path)
+                Applicability::Unsafe
             };
```

Comment:
I think I would slightly prefer something like:


```suggestion
// Unsafe when the fix would delete comments or change a used return value
let applicability = if checker.comment_ranges().intersects(range) 
  || !is_top_level_expression_call(checker, call) {
                Applicability::Unsafe
            } else {
                Applicability::Safe
};
```